Contributions/Linux_Kernel_Safety_First_Principles.md

```diff
@@ -0,0 +1,141 @@
+3. Any component within the kernel context can generate (cascaded) interference.
+4. Internal interference can cascade also through components that have been qualified for safety at unit-level.
+5. The Linux Kernel is able to interfere with both itself and any part of user-space processes (linear map).
+6. No generic solution for very complex systematic corruption of any writable memory.
+7. Stress testing alone is not sufficient for supporting safety claims.
+8. Safety claims on the Linux Kernel require both positive testing and focused negative testing.
+9. Those security features which are based on randomisation decrease repeatability of testing (e.g. structure layout randomisation).
+10. Safety claims must be supported by components with the same-or-better safety level (e.g. a safety-qualified watchdog).
+11. Non safety qualified processes can interfere with safety qualified ones, indirectly, through the kernel (e.g. triggering memory management bugs).
+12. Using cgroups/containers/SELinux removes only certain simple types of user-space-induced interference.
+13. In a mixed-criticality scenario, non safety-qualified code represents a safety liability that grows with the frequency of execution (cgroups/containers, LSM/SELinux, etc.).
+14. HW enhancements are not a catch-all solution (e.g. ECC Memory doesn't prevent interference from non-safety-qualified SW).
+15. Detection of interference depends more on the recipient than the origin of the interference: too many sources, too difficult to model reliably.
+16. Difficult to say when all necessary resources are allocated and if they will be retained (e.g. process memory pages).
+
+
+## **First Principles - Availability**
+This section provides touchstone concepts about (lack of) system availability, in presence of non-safety-qualified components.
+
+1. Detecting interference alone doesn't help with controlling/managing availability.
+2. The nature of the very complex systematic interference makes estimating probability of a failure unrealistic.
+3. Stress testing as a means of safety qualification might be realistic at most for very simple cases.
+4. Qualitatively, the likelihood of a failure grows with invocativation of non-safety-qualified components. For example:
+   1. invocation of device drivers, especially if not related to safety operations
+   2. syscalls
+   3. memory allocation/release operations
+   4. I/Os, e.g. storage, network
+   5. evaluation of Linux Security Module hooks
+   6. evaluation of cgroups tests
+   7. presence of non-safety-relevant triggering many of the above points
+
+
+## **Considerations - Toward the safety goals**
+Considerations based on the two previous sections.
+
+1. Devise negative testing simulating interference toward safety-relevant components.
+2. Rely only on external components/mechanisms that already qualified at same-o-better safety level.
+3. Analyse effects of interference based on recipient (e.g. corrupted data harder to notice)
```

Comment:
Here I would say "Make sure that the interference modes are avoided or detected either at the source or at the recipient side with sufficient level of rigor"

Reply:
No, it's intentionally excluding the source  because the whole point is that it's impossible to claim completeness about sources.

Again, if you disagree, feel free to prove me wrong and eenumerate all the possible sources.